cunumeric/deferred.py

```diff
@@ -305,37 +305,278 @@ def get_scalar_array(self):
         result = np.frombuffer(buf, dtype=self.dtype, count=1)
         return result.reshape(())
 
-    def _create_indexing_array(self, key):
-        # Convert everything into deferred arrays of int64
+    def _zip_indices(self, start_index, arrays):
+        if not isinstance(arrays, tuple):
+            raise TypeError("zip_indices expects tuple of arrays")
+        # start_index is the index from witch indices arrays are passed
+        # for example of arr[:, indx, :], start_index =1
+        if start_index == -1:
+            start_index = 0
+
+        arrays = tuple(self.runtime.to_deferred_array(a) for a in arrays)
+        # all arrays should have the same shape and type
+        data_type = arrays[0].dtype
+        if not np.issubdtype(data_type, np.integer):
```

Comment:
Besides checking that they're some kind of integer, I think you also need to cast each index array to `np.int64`, since the C++ side expects them to be of type `int64_t`.

Reply:
Moving this check to the ` _convert_key` function